modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -2623,6 +2640,242 @@ private void _validate(
 		}
 	}
 
+	private String _validateJSON(
+			CPDefinition cpDefinition, CPInstance cpInstance, String json)
+		throws PortalException {
+
+		String sanitizedJSON = json;
+
+		if (cpDefinition != null) {
+			int optionRelCount =
+				_cpDefinitionOptionRelLocalService.
+					getCPDefinitionOptionRelsCount(
+						cpDefinition.getCPDefinitionId());
+
+			if (optionRelCount > 0) {
+				JSONArray optionJSONArray = CPJSONUtil.toJSONArray(json);
+
+				for (CPDefinitionOptionRel cpDefinitionOptionRel :
+						cpDefinition.getCPDefinitionOptionRels()) {
+
+					if (cpDefinitionOptionRel.isSkuContributor()) {
+						if (cpInstance == null) {
+							throw new NoSuchCPInstanceException();
+						}
+
+						JSONArray optionRelJSONArray = CPJSONUtil.toJSONArray(
+							_cpDefinitionOptionRelLocalService.
+								getCPDefinitionOptionRelKeysCPDefinitionOptionValueRelKeys(
+									cpInstance.getCPInstanceId()));
+						boolean jsonOptionExists = false;
+
+						for (int i = 0; i < optionRelJSONArray.length(); i++) {
+							JSONObject instanceJSONObject =
```

Comment:
I don't get why this loop needs to exist inside:

```
			for (CPDefinitionOptionRel cpDefinitionOptionRel :
						cpDefinition.getCPDefinitionOptionRels()) {
```

Couldn't you just check whether instanceJSONObject.get("key") exists in a string array that only contains the key of `cpDefinition.getCPDefinitionOptionRels()`?